app/models/comment.php

```diff
@@ -20,17 +20,116 @@ public static function getComments($offset, $limit, $id)
         $db = DB::conn();
         $rows = $db->rows("SELECT * FROM comment 
                         WHERE thread_id = ? ORDER BY created ASC LIMIT {$offset}, {$limit}", array($id));
-       
+
+        foreach ($rows as $row) {
+            $favorites = array();
+            $user_favorites = $db->rows("SELECT * FROM favorites WHERE comment_id = ?", array($row['id']));
+            $total_favorites = Comment::countFavorites($row['id']);
+
+            foreach ($user_favorites as $v) {
+                $favorites[] = $v['user_id'];
+            }
+            $row['favorites'] = $favorites; 
+            $row['total_favorites'] = $total_favorites;
+            $comments[] = new Comment($row);
+        }
+        return $comments;
+    }
+
+    public static function total($thread_id)
+    {
+        $db = DB::conn();
+        return (int) $db->value("SELECT COUNT(*) FROM comment WHERE thread_id = ?", array($thread_id));
+    }
+
+    public function edit($comment_id)
+    {
+        if (!$this->body) {
+            throw new RecordNotFoundException('Record Not Found');
+        }
+        
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $db->query('UPDATE comment SET body = ?, created = NOW()
+                    WHERE id = ?', array($this->body, $comment_id));
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+            throw $e;
+        }
+    }
+
+    public static function getComment($comment_id)
+    {
+        $user = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM comment WHERE id = ?", array($comment_id));
+
+        foreach($rows as $row) {
+            $user[] = new self($row);
+        }
+        return $user;
+    }
+
+    public static function getUserComments($user_id)
+    {
+        $comments = array();
+        $db = DB::conn();
+        $rows = $db->rows("SELECT * FROM comment WHERE user_id = ?", array($user_id));
+
         foreach ($rows as $row) {
-           $comments[] = new Comment($row);
+            $comments[] = new self($row);
         }
         return $comments;
     }
 
-    public static function countComments($thread_id)
+    public static function delete($comment_id)
+    {
+        $db = DB::conn();
+        try {
+            $db->begin();
+            $db->query('DELETE FROM comment WHERE id = ?', array($comment_id));
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+            throw $e;
+        }
+    }
+
+    public static function favorites($user_id, $comment_id, $action)
+    {
+        $db = DB::conn();
+        try {
+            $db->begin();
+            if ($action === 'favorite') {
+                $db->insert('favorites', array('user_id' => $user_id, 'comment_id' => $comment_id));
+            } else {
```

Comment:
Isn't this function should be in `Favorite` class? (`favorite.php`) since its a different table (not comment)
and please separate insert and delete functionalities